jitt3/jsconf-challenge
# Nextiva challenge - JS Conf Mexico

This PR contains a solution for the Nextiva challenge - JS Conf Mexico.
```
Use Typescript to solve the following challenge: 

You are provided a string containing a list of positive integers separated by a space (" "). 
Take each value and calculate the sum of its digits, which we call it's "weight". 
Then return the list in ascending order by weight, as a string joined by a space.

For example: 99 will have "weight" 18, 100 will have "weight"
1 so in the output 100 will come before 99.

Example:
"56 65 74 100 99 68 86 180 90" ordered by numbers weights becomes: "100 180 90 56 65 74 68 86 99"

When two numbers have the same "weight", let's consider them to be strings and not numbers:
100 is before 180 because its "weight" (1) is less than the one of 180 (9)
and 180 is before 90 since, having the same "weight" (9) it comes before as a string.

All numbers in the list are positive integers and the list can be `empty.
```

**Where should the reviewer start?**
👨🏻‍💻   Go to `src/index.ts` 
🧪   For test files go to `__test__/index.test.ts` 

**How can be tested?**
:one: Install node modules (**yarn** or **npm install**)
:two: Use the test command (**yarn test**) on a terminal
:three: Test results for each file and functions will appear in the log
:four: Test cases can be updated or add new scenarios in the files inside the `__test__` folder

**Test results**
![image](https://user-images.githubusercontent.com/1787307/144188115-dc319b18-3c4b-4520-9874-4f75426ee19e.png)
